apps/src/componentLibrary/dropdown/_CustomDropdown.tsx

```diff
@@ -0,0 +1,145 @@
+import React, {
+  useCallback,
+  useMemo,
+  useRef,
+  useEffect,
+  KeyboardEvent,
+} from 'react';
+import classNames from 'classnames';
+
+import {
+  ComponentSizeXSToL,
+  DropdownColor,
+} from '@cdo/apps/componentLibrary/common/types';
+import moduleStyles from './customDropdown.module.scss';
+
+import FontAwesomeV6Icon, {
+  FontAwesomeV6IconProps,
+} from '@cdo/apps/componentLibrary/fontAwesomeV6Icon';
+import {useDropdownContext} from '@cdo/apps/componentLibrary/common/contexts/DropdownContext';
+import {dropdownColors} from '@cdo/apps/componentLibrary/common/constants';
+
+export interface CustomDropdownProps {
+  /** CheckboxDropdown name */
+  name: string;
+  /** CheckboxDropdown color */
+  color?: DropdownColor;
+  /** CheckboxDropdown size */
+  size: ComponentSizeXSToL;
+  /** CheckboxDropdown disabled state */
+  disabled?: boolean;
+  /** CheckboxDropdown label */
+  labelText: string;
+  /** Does custom dropdown hase a selected value (Renders a checkmark icon in the dropdown button if true) */
+  isSomeValueSelected?: boolean;
+  /** Custom icon to show for the dropdown button*/
+  icon?: FontAwesomeV6IconProps;
+  /** Children */
+  children: React.ReactNode;
+}
+
+/**
+ * Design System: _CustomDropdown Component.
+ * This is a PRIVATE component that is used to create different custom dropdowns.
+ * Only used inside of Design System components. DO NOT IMPORT OR USE DIRECTLY.
+ */
+const CustomDropdown: React.FunctionComponent<CustomDropdownProps> = ({
+  name,
+  labelText,
+  children,
+  isSomeValueSelected = false,
+  icon,
+  disabled = false,
+  color = dropdownColors.black,
+  size = 'm',
+}) => {
+  const {activeDropdownName, setActiveDropdownName} = useDropdownContext();
+  const dropdownRef: React.MutableRefObject<HTMLDivElement | null> =
+    useRef(null);
```

Comment:
nit: personal preference, but might be cleaner to parameterize the `useRef` and remove the type from the variable since it can be inferred, like `const dropdownRef = useRef<HTMLDivElement>(null)`